src/main/java/com/googlesps/feedon/servlets/LoginServlet.java

```diff
@@ -0,0 +1,124 @@
+   *      address. Second element contains the url for logging out
+   * 3) Logged in and registered -
+   *    - Return three elements
+   *    - First element is the user's email address
+   *    - Second element is the url for logging out
+   *    - Third element is the user's type, either a "charity" or 
+   *    - "restaurant" string
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("get method called");
+    response.setContentType("text/html");
+    ArrayList<String> result = new ArrayList<String>();
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      //check if this user has registered
+      String urlToRedirectToAfterUserLogsOut = "/index.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      result.add(userEmail);
+      result.add(logoutUrl);
+      if (hasRegistered(userEmail)) {
+        // return third element if user already registered
+        String userType = getUserType(userEmail);
+        result.add(userType);
+      }
+    } else { // not yet signed in
+      String urlToRedirectToAfterUserLogsIn = "/signin.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      result.add(loginUrl);
+    }
+    String json = new Gson().toJson(result);
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Used by user's registration form. Refer to register.html
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("post method called");
+    String name = request.getParameter("name-input");
+    String userType = request.getParameter("user-type");
+    String description = request.getParameter("description-input");
+    String location = request.getParameter("location-input");
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    Entity userEntity = new Entity("user");
+    userEntity.setProperty("email-address", userEmail);
+    userEntity.setProperty("name", name);
+    userEntity.setProperty("user-type", userType);
+    userEntity.setProperty("location", location);
+    userEntity.setProperty("description", description);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(userEntity);
+    response.sendRedirect("index.html");
+  }
+
+  /**
+   * Returns an empty string is the user is not registered, that is not 
+   * found in the datastore. Otherwise returns either type stored in the
+   * datastore, which is either "restaurant" or "charity"
+   */
+  private String getUserType(String email) {
```

Comment:
NIce job on keeping the function size small and adding java doc

Reply:
thanks!